src/accelerate/accelerator.py

```diff
@@ -608,6 +612,41 @@ def accumulate(self, model):
         with context(model):
             yield
 
+    @contextmanager
+    def join_uneven_inputs(self, joinables, even_batches=None):
+        """
+        A context manager that facilitates distributed training on uneven inputs, which acts as a wrapper around `torch.distributed.algorithms.join`.
+        """
+        if is_torch_version("<", "1.10.0"):
+            raise ValueError(f"Joining uneven inputs requires PyTorch >= 1.10.0, You have {torch.__version__}.")
+
+        if self.distributed_type == DistributedType.NO:
+            # Even when disabled, Join expects models to subclass Joinable, so skip entirely for single process runs
+            with contextlib.nullcontext(joinables):
+                yield
+
+        elif self.distributed_type == DistributedType.MULTI_GPU:
+            dl_even_batches_values = []
+
+            if even_batches is not None:
+                for dl in self._dataloaders:
+                    if not hasattr(dl, "even_batches"):
+                        raise ValueError("Overridding even_batches is not supported for iterable-style datasets")
```

Comment:
Maybe only throw an error if there are no dataloader with that attribute? I can imagine having an iterable dataset for training but a standard dataset for evaluation.

Reply:
Good spot. Based on the other feedback, I have just raised a warning for this rather than an exception